lib/shim/message-shim/subscribe-consume.js

```diff
@@ -56,13 +56,22 @@ function createSubscriberWrapper({ shim, fn, spec, destNameIsArg }) {
       }
     }
 
-    if (consumerIdx !== null) {
+    if (consumerIdx !== null && !spec.functions) {
```

Comment:
The `.functions` property should be an array in order to be valid. So we should assert that:

```suggestion
    if (consumerIdx !== null && Array.isArray(spec.functions) === false) {
```

Reply:
I can check for it being an array in the spec